decidim-blogs/app/controllers/decidim/blogs/posts_controller.rb

```diff
@@ -18,7 +18,7 @@ def show; end
       private
 
       def paginate_posts
-        @paginate_posts ||= paginate(posts.created_at_desc)
+        @paginate_posts ||= paginate(posts.published.created_at_desc)
       end
 
       def post
```

Comment:
I am able to access the blog page even though it has not yet reached its publication date.

So if you know the blog post ID, you can see it at `/processes/process-slug/f/{blog_component_id}/posts/{post_id}` if you just try the different numbers at the end of the URL.

As defined in the original issue, the blog post should not be visible on the site before the publish time has passed. If the blog post is not published yet, we should show the same kind of HTTP 404 page as we show when you try to request some unexisting URL in Decidim, e.g. try requesting https://meta.decidim.org/this_does_not_exist at MetaDecidim. Or try requesting an unexisting proposal https://meta.decidim.org/processes/roadmap/f/122/proposals/123.

Note that this error will look a bit different at development but this is what we should show in production. To get the same looking page as we get in production, change this configuration at your `config/environments/development.rb`:
```ruby
  config.consider_all_requests_local = false
```